File: evidence/generated-systems/golang/insurance-on-golang/insurance-on-golang/internal/model/Claim.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// Claim Declaration
//==============================================================
type Claim struct {
    gorm.Model
     ClaimNumber                                                            string
    NoticeDate                                                            time.Time
    LossDate                                                            time.Time
    ReportedBy                                    string
    PolicyId         *uint
    Policy           *Policy `gorm:"foreignKey:PolicyId"`
    CustomerId         *uint
    Customer           *Customer `gorm:"foreignKey:CustomerId"`
    AdjusterId         *uint
    Adjuster           *Adjuster `gorm:"foreignKey:AdjusterId"`
    IncidentId         *uint
    Incident           *Incident `gorm:"foreignKey:IncidentId"`
     Exposures           []Exposure `gorm:"foreignKey:ExposuresFromClaimId"`
     Reserves           []ClaimReserve `gorm:"foreignKey:ReservesFromClaimId"`
     ClaimPayments           []ClaimPayment `gorm:"foreignKey:ClaimPaymentsFromClaimId"`
     ServiceProviders           []ServiceProvider `gorm:"foreignKey:ServiceProvidersFromClaimId"`
     Subrogations           []SubrogationRecovery `gorm:"foreignKey:SubrogationsFromClaimId"`
    Status                      ClaimStatus
    LossCause                      CauseOfLoss

// parent associations as their child

}

File: evidence/generated-systems/angular/insurance-on-angular/src/app/models/Claim.js

// Define collection and schema for Claim
export interface Claim {
    claimNumber:
	type : ClaimNumber
    noticeDate:
	type : Date
    lossDate:
	type : Date
    reportedBy:
	type : string
    Policy:
	type : Schema.Types.ObjectId
    Customer:
	type : Schema.Types.ObjectId
    Adjuster:
	type : Schema.Types.ObjectId
    Incident:
	type : Schema.Types.ObjectId
    Exposures:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Exposure' }]
    Reserves:
 	type : [{ type: Schema.Types.ObjectId, ref: 'ClaimReserve' }]
    ClaimPayments:
 	type : [{ type: Schema.Types.ObjectId, ref: 'ClaimPayment' }]
    ServiceProviders:
 	type : [{ type: Schema.Types.ObjectId, ref: 'ServiceProvider' }]
    Subrogations:
 	type : [{ type: Schema.Types.ObjectId, ref: 'SubrogationRecovery' }]
    Status:
 	type : String
    LossCause:
 	type : String
#
    collection: 'claims'
}
